platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/module/telemetry/MapEventFactory.java

```diff
@@ -0,0 +1,139 @@
+package com.mapbox.mapboxsdk.module.telemetry;
+
+import android.content.Context;
+import android.content.res.Configuration;
+import android.net.wifi.WifiInfo;
+import android.net.wifi.WifiManager;
+import android.support.annotation.NonNull;
+import android.telephony.TelephonyManager;
+import android.text.TextUtils;
+import android.util.DisplayMetrics;
+import android.view.WindowManager;
+
+import com.mapbox.android.telemetry.Event;
+import com.mapbox.android.telemetry.TelemetryUtils;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class MapEventFactory {
+  private static final String LANDSCAPE = "Landscape";
+  private static final String PORTRAIT = "Portrait";
+  private static final String NO_CARRIER = "EMPTY_CARRIER";
+  private static final int NO_NETWORK = -1;
+
+  private static final Map<Integer, String> ORIENTATIONS = new HashMap<Integer, String>() {
+    {
+      put(Configuration.ORIENTATION_LANDSCAPE, LANDSCAPE);
+      put(Configuration.ORIENTATION_PORTRAIT, PORTRAIT);
+    }
+  };
+
+  private final Context appContext;
+
+  MapEventFactory(Context appContext) {
+   this.appContext = appContext;
+  }
+
+  Event buildMapLoadEvent() {
+    String userId = TelemetryUtils.retrieveVendorId();
+    MapLoadEvent mapLoadEvent = new MapLoadEvent(userId).setDeviceInfo(appContext);
+    mapLoadEvent.setOrientation(obtainOrientation(appContext));
+    mapLoadEvent.setAccessibilityFontScale(obtainAccessibilityFontScaleSize(appContext));
+    mapLoadEvent.setCarrier(obtainCellularCarrier(appContext));
+    mapLoadEvent.setResolution(obtainDisplayDensity(appContext));
+    mapLoadEvent.setWifi(isConnectedToWifi(appContext));
+    return mapLoadEvent;
+  }
+
+  Event buildOfflineDownloadStartEvent(String shapeForOfflineRegion,
+                                       Double minZoom, Double maxZoom,
+                                       String styleURL) {
+
+    OfflineDownloadStartEvent offlineEvent =
+      new OfflineDownloadStartEvent(shapeForOfflineRegion, minZoom, maxZoom);
+    offlineEvent.setStyleURL(styleURL);
+    return offlineEvent;
+  }
+
+  public Event buildOfflineDownloadCompleteEvent(String shapeForOfflineRegion,
```

Comment:
unused

Reply:
Will not send this event even in future?